store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -38,10 +39,44 @@ public class OwaspHtmlSanitizer implements Callable<String> {
     private String vHost;
 
     public OwaspHtmlSanitizer(String html, boolean neuterImages, String vHost) {
-        this.html = html;
+        // fix to check extra tags & remove
+        this.html = checkUnbalancedTags(html);
         this.neuterImages = neuterImages;
         this.vHost = vHost;
     }
+    
+    /** 
+     * A method to check & remove unbalanced tags which may have started, not ended
+     * 
+     * @return the Formatted Html after removing unbalanced tags
+     */
+    private String checkUnbalancedTags(String html) {
+         if (StringUtil.isNullOrEmpty(html)) {
+             return html;
+         } else {
```

Comment:
no need of else, if html is empty it will return  from if itself

Reply:
removed else block